src/opsinputs/GnssroStationIDMetOffice.cc

```diff
@@ -54,7 +54,22 @@ void GnssroStationIDMetOffice::compute(const ObsFilterData & in,
   const size_t nlocs = in.nlocs();
   std::vector<int> satid;
   in.get(Variable("MetaData/satelliteIdentifier"), satid);
-
+  
+  std::vector<int> qualityFlags;
+  in.get(Variable("MetaData/qualityFlags"), qualityFlags);
+  std::vector<int> isRising(nlocs, 2);
+  for (size_t i=0; i < nlocs; ++i) {
+    if ((qualityFlags[i] & (1 << 13)) != 0) {
+      isRising[i] = 1;
+    } else if ((qualityFlags[i] & (1 << 13)) == 0) {
+      isRising[i] = 0;
+    }
+    else {
```

Comment:
Unless I have read the code incorrectly I don't think you can ever reach this `else` block. How about setting `isRising` to `0` by default, then having a single `if` block that sets it to `1` if the first condition is met.

Reply:
Makes sense. If there is a chance it could be missing, checking for that would be a good idea prior to checking its 13th bit. Having said that, looking at OPS, it checks if the variable exists at all - which might be equivalent to failing to retrieve it from the obsspace?